pkg/controllers/cloud/cloudprovideraccount_controller.go

```diff
@@ -214,3 +240,87 @@ func (r *CloudProviderAccountReconciler) getAccountProviderType(namespacedName *
 
 	return r.accountProviderType[*namespacedName]
 }
+
+// getCPABySecret returns nil only when the Secret is not used by any CloudProvideAccount CR,
+// otherwise the dependent CloudProvideAccount CR will be returned.
+func (r *CloudProviderAccountReconciler) getCPABySecret(s types.NamespacedName) (error, []cloudv1alpha1.CloudProviderAccount) {
+	cpaList := &cloudv1alpha1.CloudProviderAccountList{}
+	err := r.Client.List(context.TODO(), cpaList, &client.ListOptions{})
+	var cpaItems []cloudv1alpha1.CloudProviderAccount
+	if err != nil {
+		return fmt.Errorf("failed to get CloudProviderAccount list, err:%v", err), nil
+	}
+
+	for _, cpa := range cpaList.Items {
+		if cpa.Spec.AWSConfig != nil {
+			if cpa.Spec.AWSConfig.SecretRef.Name == s.Name &&
+				cpa.Spec.AWSConfig.SecretRef.Namespace == s.Namespace {
+				cpaItems = append(cpaItems, cpa)
+			}
+		}
+
+		if cpa.Spec.AzureConfig != nil {
+			if cpa.Spec.AzureConfig.SecretRef.Name == s.Name &&
+				cpa.Spec.AzureConfig.SecretRef.Namespace == s.Namespace {
+				cpaItems = append(cpaItems, cpa)
+			}
+		}
+	}
+	return nil, cpaItems
+}
+
+// watchSecret watch the Secret objects.
+func (r *CloudProviderAccountReconciler) watchSecret() error {
+	for {
+		event, ok := <-r.watcher.ResultChan()
+		if !ok {
+			r.resetSecretWatcher()
+		} else {
+			switch event.Type {
+			case watch.Modified:
+				secret := event.Object.(*v1.Secret)
+				namespacedName := types.NamespacedName{Namespace: secret.Namespace, Name: secret.Name}
+				err, cpaItems := r.getCPABySecret(namespacedName)
+				if err != nil {
+					r.Log.Error(err, "error while getting CPA by Secret %v", secret.Name)
```

Comment:
For consistent logging(Similiar to L297), can you dump the secret NamespacedName instead of Name?

Reply:
Solved.